### PR TITLE
feat(util): add get_docs_batch_generator for documentarray

### LINKED ISSUE #3333: Add  `get_docs_batch_generator` for DocumentArray
This is probably the most used function from jina commons. I think it would make sense to add it to jina - maybe not as a DocumentArray method, but as a util somewhere. Would enable us to eliminate jina commons requirement from a lot of executors

### PR DESCRIPTION
**Pull request by agarwalrounak**
Closes #3333 

This PR aims to add `get_docs_batch_generator` for DocumentArray. This change would enable us to eliminate jina commons requirement from a lot of executors. 

The existing import statement `from jina_commons.batching import get_docs_batch_generator` in executors can be updated to `from jina.util import get_docs_batch_generator`. Another way could be to add `from jina.util import get_docs_batch_generator` to `jina/__init__.py` file and update the existing import statement in executors to `from jina import get_docs_batch_generator`.

### CODE REVIEW

**Review by hanxiao — CHANGES_REQUESTED** [review]
`.batch` should be a method of `DocumentArray` & `DocumentArrayMemap`

**Comment by agarwalrounak** [comment]
Hi @hanxiao, thank you for your review! 
I had a query. By _**`.batch` should be a method of `DocumentArray` & `DocumentArrayMemap`**_, do you mean that add a method `batch` to the classes `DocumentArray` & `DocumentArrayMemap`, which then calls the `get_docs_batch_generator` method added in `util/__init__.py`? Please let me know if I understood it correctly.

**Comment by hanxiao** [comment]
> By doing so, we would then be able to change the occurrences of get_docs_batch_generator(...) in executors to DocumentArray.batch(...) or DocumentArrayMemap.batch(...)

Yes, that would be the user experience.

I would suggest to implement `.batch` in `jina/types/arrays/traversable.py:TraversableSequence` mixin, as this mixin class is shared by both `DocumentArray` and `DocumentArrayMemap`.

**Comment by agarwalrounak** [comment]
Thank you for your suggestion. I have implemented `batch` method in `TraversableSequence` mixin class. `batch` calls `get_docs_batch_generator` method added in `util/__init__.py` and returns the `Generator` created.

File: jina/types/arrays/traversable.py [hunk 1]
```diff
@@ -95,3 +97,22 @@ def traverse_flat(self, traversal_paths: Iterable[str]) -> 'TraversableSequence'
     @abstractmethod
     def _flatten(sequence):
         ...
+
+    @staticmethod
```
Comment (hanxiao) [inline jina/types/arrays/traversable.py:101]: can you simply move `get_docs_batch_generator` function into `.batch` and make `.batch` **not** staticmethod?

The reason is that `DocumentArray` to jina is like `ndarray` to numpy, it is our basic data type, and we'd like to have any method that relate to `DocumentArray` or `DocumentArrayMemmap` as its "native" class method. We do not prefer "misc"/"util" module to store orphan methods, it is not part of our design. In Jina, every method should have a clear parent (at least to the best reasonable effort).

File: jina/util/__init__.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+"""Module for Jina helper functions for batching documents in executors."""
```
Comment (hanxiao) [inline jina/util/__init__.py:1]: Delete this file.

In general, we do not prefer some "misc"/"util" module to store orphan methods, it is not part of our design. In Jina, every method should have a clear parent (at least to the best reasonable effort).

**Review by hanxiao — CHANGES_REQUESTED** [review]
- move `get_docs_batch_generator` inside `jina/types/arrays/traversable.py` and **un-**staticmethod the `.batch` func
- need unit tests

**Comment by agarwalrounak** [comment]
@hanxiao I was away from my laptop for a while, so I couldn't work on the changes you suggested. Sorry for that! I see that you opened a new PR with the changes. 
I will pick up a different issue to work on.

**Pull request closed without merging**